documentation/contributing/index.markdown

```diff
@@ -0,0 +1,108 @@
+
+As an aside, a lot of documentation exists as Q&As on there, and if you feel so compelled, help us move some of that documentation onto the website.
+
+## Reporting & Fixing Bugs
+
+Bugs should be reported to the Github issues page of the appropriate repo. Please leave as much detail as possible including ROS and Ubuntu version. Note however that we have limited resources to fix individual bugs, and are more inclined to give advice on helping you fix your own bugs.
+
+### Pull Requests
+
+Our policy for contributing changes to the code base is that no one, not even the maintainers, should commit directly to the repo. Rather, you should create a feature branch on your own fork of the project then create a PR when it's ready to be reviewed by someone else. All changes should be merged by someone other than the author of the PR. The only exception to this rule is when cherry-picking commits from older branches (previous ROS releases) to newer branches.
+
+## Reviewing Pull Requests
+
+Anyone can review open pull requests and comment on the changes. While only a handful of developers have push access to the repositories, your +1 in the comment threads helps us merge changes with more confidence and quickness.
+
+To make it easier to review pull requests from others, we are currently working on [merging several repos togther](http://discourse.ros.org/t/migration-to-one-github-repo-for-moveit/266)
+
+### Running Tests Locally
+
+You can run MoveIt! tests locally with catkin-tools. To run a test for just 1 package:
+
+    catkin run_tests --no-deps --this -iv
+
+Run test for all packages:
+
+    catkin run_tests -iv
+
+### Using Docker to Review Pull Requests
+
+Docker can help you easily evaluate someone else's changes without breaking your local setup, as well as test on other versions of Linux other than your local installed version. MoveIt! now has an [official MoveIt! Docker build](https://hub.docker.com/r/davetcoleman/moveit_docker/) that lets you quickly get a MoveIt! setup in a local container. Once docker is setup on your system, all you really need is:
+
+    docker run -it moveit/moveit_docker:moveit-kinetic-source
+
+Though there are many other images available for various ROS distros and purposes.
+
+## Adding New Features
+
+We would like to see MoveIt! continue to evolve and so new features are encouraged. However, it can be difficult to ensure that new features do not break existing functionality or user's code. For this reason, we ask you keep your pull requests in the smallest digestible chunks possible - not thousands of lines of changed code.
+
+We recommend you open an issue first asking if the change is a good idea, before implementing it, so that you don't waste your time.
+
+### Unit/Integration Tests
+
+Unit tests and integration tests are **always** welcomed, please add them to your PRs. TODO: add more information here about how to create tests
```

Comment:
Similar to my note above, I think tests should be a separate topic. I've removed my "TODO" note - instead we should add a link to a different page that goes into greater details on that. Would you mind making that new page and copying your notes here as a starter?